src/vignette/util/image_response.clj

```diff
@@ -38,32 +43,39 @@
 
 (defn create-image-response
   ([image image-map]
-   (-> (response (->response-object image))
-       (header "Content-Type" (content-type image))
-       (header "Content-Length" (content-length image))
-       (header "ETag" (etag image))
-       (header "X-Thumbnailer" "Vignette")
-       (add-content-disposition-header image-map)
-       (add-surrogate-header image-map)))
+   (as-> (response (->response-object image)) resp
+         (when-header-val resp "Content-Type" (content-type image))
+         (when-header-val resp "Content-Length" (content-length image))
+         (when-header-val resp "ETag" (etag image))
+         (header resp "X-Thumbnailer" "Vignette")
+         (add-content-disposition-header resp image-map image)
+         (add-surrogate-header resp image-map)))
   ([image]
     (create-image-response image nil)))
 
 (defn create-head-response
   [image-map]
   (-> (response "")
-       ;(header "Content-Type" (content-type image))
        (header "X-Thumbnailer" "Vignette")
        (add-content-disposition-header image-map)
        (add-surrogate-header image-map)))
 
+(defn- base-filename [image-map object]
```

Comment:
What is the filename in the case of the SAS request? Can it be derived from the image map?

If not it seems like this is something that should be handled with polymorphism. That is we should create a protocol and add the `filename` to the `image-map`.